python/sglang/srt/models/qwen2.py

```diff
@@ -283,11 +285,15 @@ def forward(
         positions: torch.Tensor,
         input_metadata: InputMetadata,
         input_embeds: torch.Tensor = None,
+        get_embedding: bool = False,
     ) -> torch.Tensor:
         hidden_states = self.model(input_ids, positions, input_metadata, input_embeds)
-        return self.logits_processor(
-            input_ids, hidden_states, self.lm_head.weight, input_metadata
-        )
+        if get_embedding:
+            return self.pooler(hidden_states, input_metadata)
+        else:
+            return self.logits_processor(
+                input_ids, hidden_states, self.lm_head.weight, input_metadata
+            )
```

Comment:
Change the order of two branches. We always choose the generation branch as the first one.